text/0000-partial_types.md

```diff
@@ -0,0 +1,672 @@
+
+Partial Types is a **minimal** and full extension to the Rust Type System, which allows to safe control easily partial parameters and all kinds of partial not consumption.
+
+Advantages: maximum type safety, maximum type control guarantee, no ambiguities, flexibility, usability and universality.
+
+
+# Motivation
+[motivation]: #motivation
+
+Safe, Flexible controllable partial parameters for functions and partial not consumption (including partial not borrowing) are highly needed and this feature unlock huge amount of possibilities.
+
+Partial borrowing is already possible in Rust, as partial referencing and partial moves.
+
+But partial parameters are forbidden now, as qualified consumption: partial not borrowing, partial not referencing, partial not moving and partial initializing.
+
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+_**Note**: I didn't comment type explanations (as is needed for compiler) for colorizing purposes only._
+
+_**Note**: I use symbol `~` in code as a synonym of "equivalent" word only. It is not a Rust operator!_
+
+
+## Partial types by type Integrity
+
+```rust
+// case (A1)
+let foo : i16 = 0;
+```
+
+What is a **full type** of `foo`? Is it the `i16`? No, the `i16` is a sub-full_type in Rust type-system.
+```rust
+// case (A2)
+<variable>  : <full_type>;
+<variable>  : <type_clarification> <type>;
+<variable>  : <sharing> <'lifetime> <mutability> <type>;
+```
+
+If we explicitly write **full type** (using _unused_ Rust names) we get:
+```rust
+// case (A3)
+// FROM case (A1)
+foo  : value 'static const i16;
+```
+
+That mean, that variable `foo` has the `i16` (sub-)type with next type-clarification: `const`(not `mut`) mutability, `value`(not `&`) sharing and `'static` lifetime.
+
+I propose to extend type system by adding type integrity to sub-type. So, our variable will have next full type:
+```rust
+// case (A4)
+<variable>  : <type_clarification> <%integrity> <type>;
+<variable>  : <sharing> <'lifetime> <mutability> <%integrity> <type>;
+
+// case (A5)
+// FROM case (A1)
+foo  : value 'b const %a i16;
+foo  : value 'static const %full i16;
+foo  : %full i16;
+foo  : i16;
+```
+
+Lifetime variants are `'static` (for static lifetime), `'_`(don't care lifetime) and any other `'b`(some "b" lifetime).
+
+By the same analogy, integrity has similar names and meanings: `%full`(full integrity, soft keyword), `%_`(don't care how partial integrity is, soft keyword), `%empty` or `%!` (no integrity, soft keyword) and any other `%a`(some "a" integrity).
+
+Symbol `%` percent mean percent or part of the whole thing (variable in our case).
+
+_Note_: It is highly recommended to deprecate operator `%` as a remainder function (it is still no ambiguities to write "`\s+%\s+`"), and replace it with another operator (for example: `%mod` / `%rem` / `mod` / `rem`) to not to be confused by type integrity. 
```

Comment:
After reading this entire section, I have no idea what you mean by "integrity." I'm not sure if English is your first language, but to me, levels of integrity don't even make sense for a unit like `i16` since it's a primitive type. You could say that the value has no integrity, but that would mean something like it being uninitialized, not it being partially owned.

Reply:
Really? Sorry, I've change type "integrity" to type "access", even if it look not so accurate to me.